### PR TITLE
TEST/PERF: Reduce testing time for test_ucp_wait_mem

### PR DESCRIPTION
**Pull request by yosefe**
## Why
Reduce overall testing time

### CODE REVIEW

File: test/gtest/ucp/test_ucp_perf.cc [hunk 1]
```diff
@@ -358,7 +358,16 @@ UCS_TEST_SKIP_COND_P(test_ucp_perf, envelope, has_transport("self"))
 UCP_INSTANTIATE_TEST_CASE(test_ucp_perf)
 
 
-class test_ucp_wait_mem : public test_ucp_perf {};
+class test_ucp_wait_mem : public test_ucp_perf {
+public:
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        // Add test instance only if wait_mem is non-trivial for this arch
+        if (ucs_arch_wait_mem != ucs_arch_generic_wait_mem) {
```
Comment (hoopoepg) [inline test/gtest/ucp/test_ucp_perf.cc:366]: hacky way :)

**Review by hoopoepg — APPROVED** [review]

File: test/gtest/ucp/test_ucp_perf.cc (hunk 1 above)
Comment (shamisp) [reply test/gtest/ucp/test_ucp_perf.cc]: yeah...this is just asking for a trouble...

Comment (yosefe) [reply test/gtest/ucp/test_ucp_perf.cc]: @shamisp is it better now? changed to check arch

Comment (shamisp) [reply test/gtest/ucp/test_ucp_perf.cc]: yes, thanks

**Review by hoopoepg — APPROVED** [review]

**Pull request merged**